[angsongyee] iP

angsongyee opened the pull request:
# SHODAN
> "How can you challenge a perfect, immortal machine?" &mdash; [SHODAN](https://systemshock-archive.fandom.com/wiki/SHODAN)

SHODAN is a bot that helps manage your tasks! 🤖

You can:
- Save _varied_ types of tasks
- Search for them easily
- ~~Delete your tasks~~

Reasons to use SHODAN:
1. 👽
2. 🚀

Features:
- [x] Managing tasks
- [ ] Destroying all sentient life (coming soon)

## Running SHODAN
Running SHODAN is simple, you just need to have this in your `main` method.
```java
public class Main {
   public static void main(String[] args) {
      Shodan chatbot = new Shodan();
      chatbot.run();
   }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

angsongyee committed:
Updated to Level 0 (Rename, Greet, Exit)

angsongyee committed:
add Level 1 (Echo)

angsongyee committed:
add Level 2 (Add, List)

angsongyee committed:
add Level 3 (Mark As Done)

angsongyee committed:
add Level 4 (ToDos, Events, Deadlines)

angsongyee committed:
Add text ui testing

angsongyee committed:
add Level 5 (Handle errors)

angsongyee committed:
add Level 6 (Delete)

angsongyee committed:
update text-ui-tests for new delete command and exception messages

angsongyee committed:
Update switch-case formatting to match style guide

angsongyee committed:
Add save and load for tasks

Shodan now saves the current list of tasks to the filesystem
whenever a change is made. Upon startup, it also tries to find
to find the saved tasks and load them. This allows the tasks to
persist between sessions.

angsongyee committed:
Modified tasks to be able to store date and time

angsongyee committed:
Move parsing, storage, tasks, ui into own classes

In addition to the new classes, the entire project
has been moved to the shodan package.

angsongyee committed:
Update text-ui-test configs to match changes

angsongyee committed:
Merge branch 'add-gradle-support'

angsongyee committed:
Add support for gradle builds

angsongyee committed:
Add JUnit testing, tests for TaskSerializer

JiaXinEu commented on src/main/java/shodan/Shodan.java:
```diff
@@ -0,0 +1,53 @@
+import java.util.StringJoiner;
+
+public class Shodan {
+    private TaskList tasks;
```
I like how you use plural noun to distinguish clearly between multi-valued variables

JiaXinEu commented on src/main/java/shodan/tasks/TaskParser.java:
```diff
@@ -0,0 +1,99 @@
+        StringJoiner startDate = new StringJoiner(" ");
+        StringJoiner endDate = new StringJoiner(" ");
+        int state = 0;
+        int sections = 1;
```
Should this be singular?

angsongyee replied:
Thanks for catching that, I'll rename it to something more suitable :smile:

JiaXinEu commented on src/main/java/shodan/storage/StorageManager.java:
```diff
@@ -0,0 +1,50 @@
+import java.util.List;
+import java.util.ResourceBundle;
+
+public class StorageManager {
```
I like how you use noun for class name 😃

JiaXinEu commented on src/main/java/shodan/tasks/Task.java:
```diff
@@ -0,0 +1,34 @@
+
+public abstract class Task {
+    private String name;
+    private boolean isDone = false;
```
I like how the naming of the Boolean variable is at sufficient level of detail

JiaXinEu reviewed (commented):
LGTM 👍 
Overall, I think the code is easy to read and almost no ambiguities regarding the intent of a variable or a method.

angsongyee committed:
Updated variable naming to follow the style guide

angsongyee committed:
Add JavaDoc comments

angsongyee committed:
Update naming, line length to follow style guide

angsongyee committed:
Add find command to search for tasks

angsongyee committed:
Merge branch 'branch-A-JavaDoc'

angsongyee committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/shodan/tasks/Task.java
#	src/main/java/shodan/tasks/TaskParser.java
#	src/main/java/shodan/tasks/impl/Event.java

angsongyee committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/shodan/TaskList.java
#	src/main/java/shodan/ui/TermUi.java

angsongyee committed:
Update README.md

angsongyee committed:
Remove all unnecessary usages of 'this'

ronnnnnnnnn commented on src/main/java/shodan/command/impl/MarkCommand.java:
```diff
@@ -0,0 +1,44 @@
+    public MarkCommand(int index, boolean toMark) {
+        this.index = index;
+        this.toMark = toMark;
+    }
```
Should a line break be added here?

ronnnnnnnnn commented on src/main/java/shodan/command/impl/AddCommand.java:
```diff
@@ -0,0 +1,39 @@
+    /**
+     * Instantiates a new Add command.
+     *
+     * @param args     the list of arguments passed in
+     * @param taskType the type of the task to add.
```
I like the even spacing between the parameter name and description.

ronnnnnnnnn reviewed (commented):
LGTM, overall code and formatting are easy to read. No major coding standard violations.

angsongyee committed:
Add CheckStyle to project

angsongyee committed:
Fix formatting issues and add missing JavaDoc

angsongyee committed:
Add GUI to project, set default launch to GUI

angsongyee committed:
Fix gradle config to build fat jars correctly

angsongyee committed:
Fix error message appearing when loading save

angsongyee committed:
Add assertions

angsongyee committed:
Refactor code to improve code quality

angsongyee committed:
Merge pull request #3 from angsongyee/branch-A-Assertions

Add assertions

angsongyee committed:
Merge branch 'master' into branch-A-CodeQuality

angsongyee committed:
Merge pull request #2 from angsongyee/branch-A-CodeQuality

Refactor code to improve code quality

angsongyee committed:
Implement C-BetterSearch

The find command now supports multiple keyword search,
matching partial keywords, and searching by task type
(e.g. `find event` will list all events in the task list).

Note that all 3 features can be used in a single search,
for example `find todo book` will list all todo tasks and
all tasks that have "book" in their name.

angsongyee committed:
Add dark theme, ui improvements

Added dark theme using CSS, and tweaked the spacing
between messages.

angsongyee committed:
Add product screenshot for website

angsongyee committed:
Update README.md for User Guide Website

angsongyee committed:
Renamed UI.png to Ui.png

angsongyee committed:
Merge remote-tracking branch 'origin'

angsongyee committed:
Update screenshot file name in README.md
